fvm/fvm_bench_test.go

```diff
@@ -0,0 +1,1443 @@
+)
+
+type vmTestContext struct {
+	chain                   flow.Chain
+	serviceAccountSeqNumber uint64
+}
+
+func (c *vmTestContext) serviceAccountSequenceNumber() uint64 {
+	c.serviceAccountSeqNumber = c.serviceAccountSeqNumber + 1
+	return c.serviceAccountSeqNumber - 1
+}
+
+// BenchmarkRuntimeEmptyTransaction simulates executing blocks with `transactionsPerBlock`
+// where each transaction is an empty transaction
+func BenchmarkRuntimeTransaction(b *testing.B) {
+	transactionsPerBlock := 10
+
+	chain := flow.Testnet.Chain()
+
+	benchTransaction := func(b *testing.B, tx string) {
+		tctx := &vmTestContext{
+			chain: flow.Testnet.Chain(),
+		}
+
+		executeBlocks := prepareExecutionEnv(b, tctx.chain)
+
+		// Create an account private key.
+		privateKeys, err := testutil.GenerateAccountPrivateKeys(1)
+		require.NoError(b, err)
+
+		// Bootstrap a ledger, creating accounts with the provided private keys and the root account.
+		accounts, err := createManyAccounts(b, tctx, executeBlocks, privateKeys)
+		require.NoError(b, err)
+
+		deployContract(b, tctx, executeBlocks, accounts[0], privateKeys[0], "TestContract", `
+access(all) contract TestContract {
+	access(all) event SomeEvent()
+
+    access(all) fun empty() {
+    }
+
+	access(all) fun emit() {
+		emit SomeEvent()
+    }
+}
+`)
+
+		btx := []byte(tx)
+
+		b.ResetTimer() // setup done, lets start measuring
+		for i := 0; i < b.N; i++ {
+			transactions := make([]*flow.TransactionBody, transactionsPerBlock)
+			for j := 0; j < transactionsPerBlock; j++ {
+
+				txBody := flow.NewTransactionBody().
+					SetScript(btx).
+					AddAuthorizer(tctx.chain.ServiceAddress()).
+					SetProposalKey(tctx.chain.ServiceAddress(), 0, tctx.serviceAccountSequenceNumber()).
+					SetPayer(tctx.chain.ServiceAddress())
+
+				err := testutil.SignEnvelope(txBody, tctx.chain.ServiceAddress(), unittest.ServiceAccountPrivateKey)
+				require.NoError(b, err)
+
+				transactions[j] = txBody
+			}
+
+			computationResult := executeBlocks([][]*flow.TransactionBody{transactions})
+			for j := 0; j < transactionsPerBlock; j++ {
+				require.Empty(b, computationResult.TransactionResults[j].ErrorMessage)
+			}
+		}
+	}
+
+	templateTx := func(prepare string) string {
+		return fmt.Sprintf(`
+import FungibleToken from 0x%s
+import FlowToken from 0x%s
+import TestContract from 0x%s
+
+transaction(){
+	prepare(signer: AuthAccount){
+		var i = 0
+		while i < 100 {
+			i = i + 1
+%s
+		}
+	}
+}`, fvm.FungibleTokenAddress(chain), fvm.FlowTokenAddress(chain), "754aed9de6197641", prepare)
+	}
+
+	b.Run("reference tx", func(b *testing.B) {
+		benchTransaction(b, templateTx(""))
+	})
+	b.Run("log empty string", func(b *testing.B) {
+		benchTransaction(b, templateTx(`log("")`))
+	})
+	b.Run("log long string", func(b *testing.B) {
+		benchTransaction(b, templateTx(`log("0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000")`))
+		// 100 character string
+	})
```

Comment:
not sure if we care that much about these methods from benchmarking perspective,

Reply:
in case if people leave them in transactions, It could be good to know if they have a significant performance cost.

Reply:
you're right but I think we ignore them anyway for transactions and won't even capture them in the first place.